lib/beautiful/xresources.lua

```diff
@@ -103,7 +103,11 @@ end
 -- @tparam[opt] integer|screen s The screen.
 -- @treturn integer Resulting size (rounded to integer).
 function xresources.apply_dpi(size, s)
-    return round(size / 96 * xresources.get_dpi(s))
+    if s then
+        return s:dpi_scale(size)
+    else
+        return round(size * require("beautiful").screen_font_dpi()/96)
```

Comment:
Why does this use font DPI? If you just use the font DPI everywhere, then `s.dpi` is not needed.

Reply:
There's an issue about “what has this call being used for in legacy code” and “how do we transition from there to the API we want now”.

Among other things, this means that the only case where you access the DPI without a screen specification is if you want to know the user-specified font scaling, since there is (or should not be) any concept of “global DPI” (otherwise).